test/integration/main_test.go

```diff
@@ -334,33 +347,39 @@ deletion_timeline:
 
 // testGetFirstPmapEventWithRetries queries the DB and get the matched PmapEvent.
 // If no matched PmapEvent was found, the query will be retried with the specified retry inputs.
-func testGetFirstPmapEventWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *config) *v1alpha1.PmapEvent {
+func testGetFirstPmapEventWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *config) (*v1alpha1.PmapEvent, string) {
```

Comment:
I'd suggest doing some refactor here. Use one function to get the BQ rows with retries and then write another two function to get pmap event and processError. Currently the signature looks confusing, the naming is `testGetFirstPmapEventWithRetries` and the return is one pmapevent and one string(what does the return string means is confusing here)

Reply:
renamed to `testGetFirstBQEntryWithRetires`, and now this returns a `bqEntry`